src/client.rs

```diff
@@ -107,70 +137,82 @@ impl Client {
     /// server remembers the state of the subscription. This allows for many
     /// different modes of operations compared to a regular or catchup
     /// subscription where the client holds the subscription state.
-    pub fn create_persistent_subscription<S>(
+    pub async fn create_persistent_subscription<StreamName, GroupName>(
```

Comment:
**CONCERN:** This is probably out of the scope of this PR, but as I know, the general conclusion was to separate the streams and management functionalities (so e.g. introduce separate client classes).

Reply:
it is but it was about separating operational  (related DB management) from data access commands.